src/api/status/public/js/serviceStatus.js

```diff
@@ -0,0 +1,65 @@
+const reportTbody = () => document.querySelector('#report');
+const requestMessageReport = () => document.querySelector('#request-message');
+
+// Create td for service name
+const createTdText = (content) => {
+  const td = document.createElement('td');
+  const div = document.createElement('div');
+  const h6 = document.createElement('h6');
+
+  h6.className = 'text-capitalize mb-0 text-sm';
+  h6.innerHTML = content;
+  div.className = 'text-start px-2 py-1';
+  div.appendChild(h6);
+  td.appendChild(div);
+
+  return td;
+};
+
+// Create td for icon
+const createTdIcon = (status) => {
+  const td = document.createElement('td');
+  const icon = document.createElement('i');
+
+  td.className = 'align-middle text-center';
+  icon.title = status;
+  icon.className =
+    status >= 200 && status <= 299
+      ? 'text-lg text-success fas fa-check-circle'
+      : 'text-lg text-danger fas fa-exclamation-circle';
+
+  td.appendChild(icon);
+
+  return td;
+};
+
+// Create row of the table
+const serviceRowFormat = (result) => {
+  const tr = document.createElement('tr');
+
+  tr.appendChild(createTdText(result.name));
+  tr.appendChild(createTdIcon(result.status.staging));
+  tr.appendChild(createTdIcon(result.status.production));
+  reportTbody().appendChild(tr);
+};
+
+export default function getAllServicesStatus() {
+  fetch('/v1/status/status')
+    .then((res) => {
+      if (!res.ok) {
+        requestMessageReport().innerHTML = '';
+        const icon = document.createElement('i');
+        icon.className = 'fas fa-server px-2';
+        requestMessageReport().appendChild(icon);
+        requestMessageReport().innerHTML += `${res.status} Unable to get API status.`;
+        throw new Error('unable to get API status');
+      }
+      return res.json();
+    })
```

Comment:
You should exit early and reuse variable.
```suggestion
      if (res.ok) {
        return res.json();
      }
        const messageReport = requestMessageReport();
        if (messageReport) {
          messageReport.innerHTML = '';
          const icon = document.createElement('i');
          icon.className = 'fas fa-server px-2';
          messageReport.appendChild(icon);
          messageReport.innerHTML += `${res.status} Unable to get API status.`;
          throw new Error('unable to get API status');
        }
    })
```

Reply:
Thanks, changed.
I don't think we need a check for messageReport.

Reply:
It's a safe error handling because `querySelector` can return Element | null. I got used to Typescript strictness.

Reply:
If so, I have to add a empty `return;` after the if, since then() required either an return or throw

Reply:
nit:
```suggestion
if (messageReport) {
          messageReport.innerHTML = '';
          const icon = document.createElement('i');
          icon.className = 'fas fa-server px-2';
          messageReport.appendChild(icon);
          messageReport.innerHTML += `${res.status} Unable to get API status.`;
        }
 throw new Error('unable to get API status');
```
@Kevan-Y

Reply:
Fixed